### DISCUSSION TITLE
Dynamically query for window ID, position, or size?

### DISCUSSION THREAD

**Discussion by j-hui** [nvim-tree/nvim-tree.lua#2532] [General]
I'm trying to see if I can get Fidget to avoid rendering on top of the nvim-tree window when it's open. I see there's already an event subscription mechanism, but is there any way to dynamically query the position and width of the nvim-tree window?

I saw [this question](https://github.com/nvim-tree/nvim-tree.lua/discussions/1813) but it suggests the only option is to enumerate through all window IDs and search for the nvim-tree window. Is there any API that is more direct?

Thanks!

**Comment by alex-courtis** [comment]
We did add public API: `:help nvim-tree-api.tree.is_tree_buf()` however that doesn't get you much closer.

How about API something like 
```help
tree.winid({opts})                              *nvim-tree-api.tree.winid()*
    Retrieve the winid of the open tree.

    Parameters: ~
      • {opts} (table) optional parameters

    Options: ~
      • {tabpage}        (number)  tabpage, 0 for current, default 0

    Return: ~
      (number) winid or nil if tree is not visible
```

**Reply by j-hui** [reply]
An API like `tree.winid()` would be fantastic.

**Reply by alex-courtis** [reply]
#2544

As always, [Pull Request](https://github.com/nvim-tree/nvim-tree.lua/wiki/Development) are most gratefully appreciated.

**Reply by j-hui** [reply]
I’ll give it a shot!
